### PR TITLE
Cat fixes round 2

### PR DESCRIPTION
**Pull request by Theundyingcode**
#### Summary

```SUMMARY: Bugfixes "Fix weird item categorization part 2"```
#### Purpose of change

closes: #26221
#### Describe the solution

Notable Changes:

- Poop is no longer food it's chemical stuff
- Changed weapon "balls"  to other like the rest of them
- Tools are tools not weapons
- Methcola is a drug
- Tainted tornado is mutagen
- Canteens are clothing like all the other wearable equipment
- Glue is a spare part
- Medieval_Stuff/shields are all armor

#### Additional context

This finishes fixing all the items on my fix now list except blazmod items.  From what I can tell, all of blaze_blob.json are tools.

### CODE REVIEW

File: data/json/items/generic.json [hunk 1]
```diff
@@ -2754,5 +2758,36 @@
     "material": "plastic",
     "symbol": ")",
     "color": "green"
+  },
+  {
+    "type":"GENERIC",
+    "id" : "sheet_metal",
+    "name" : "sheet metal",
```
Comment (mlangsdorf) [inline data/json/items/generic.json]: what's wrong with the definition of sheet_metal in data/json/items/vehicle_parts.json?  Why is this repeated here?  I think changing the category makes sense, I think moving the definition around doesn't.

File: data/json/items/generic.json [hunk 2]
```diff
@@ -2754,5 +2758,36 @@
...
+    "description" : "A thin sheet of metal.",
+    "//" : "Roughly in the 70cm×70cm×1.5mm to 1m×1m×0.75mm range. Has to be folded / rolled up when in inventory, so 75% density compared to solid block.",
+    "weight" : 6000,
+    "to_hit" : -2,
+    "color" : "light_cyan",
+    "symbol" : "]",
+    "material" : ["steel"],
+    "volume" : 4,
+    "bashing" : 5,
+    "category" : "spare_parts",
```
Comment (mlangsdorf) [inline data/json/items/generic.json]: Consider moving category up, to around name and id, in line with other definitions.

File: data/json/items/generic.json [hunk 3]
```diff
@@ -2730,7 +2734,7 @@
   {
     "id": "clay_pot_flower",
     "type": "GENERIC",
-    "category": "tools",
+    "category": "other",
```
Comment (mlangsdorf) [inline data/json/items/generic.json]: Putting stuff back into "other" seems like a step backwards.  They don't seem to have any uses, so "salvage" instead?  Alternately, "construction materials", "containers", or "deployables".

File: data/json/items/melee.json [hunk 4]
```diff
@@ -277,6 +277,7 @@
     "symbol": "*",
     "color": "white",
     "name": "baseball",
+    "category": "other",
```
Comment (mlangsdorf) [inline data/json/items/melee.json]: weapons, not other.

File: data/json/items/melee.json [hunk 5]
```diff
@@ -305,6 +307,7 @@
     "symbol": "*",
     "color": "brown",
     "name": "football",
+    "category": "other",
```
Comment (mlangsdorf) [inline data/json/items/melee.json]: "salvage", not "other"

File: data/json/items/melee.json [hunk 6]
```diff
@@ -1770,6 +1775,7 @@
     "type": "GENERIC",
     "id": "pitchfork",
     "name": "pitchfork",
+    "category": "tools",
```
Comment (mlangsdorf) [inline data/json/items/melee.json]: "weapons", since you can't actually use it to pitch hay.

File: data/json/items/generic.json (hunk 3 above)
Comment (Theundyingcode) [reply data/json/items/generic.json]: The categories are not in yet and this bugs me.

File: data/json/items/melee.json (hunk 5 above)
Comment (Theundyingcode) [reply data/json/items/melee.json]: The categories are not in yet...

**Comment by Theundyingcode** [comment]
This is just a stopgap measure to address some of the glaring bad things.   Since everything is going to need to get recategorized after the new categories get added.

**Pull request merged**